app/hotels/src/allHotels/AllHotelsSearchList.js

```diff
@@ -2,38 +2,102 @@
 
 import * as React from 'react';
 import idx from 'idx';
-import { createFragmentContainer, graphql } from 'react-relay';
-import { ScrollView, Text } from 'react-native';
-import { CenteredView, Logger } from '@kiwicom/react-native-app-shared';
+import { createPaginationContainer, graphql } from 'react-relay';
+import type { Relay } from 'react-relay';
+import { Text, FlatList, View, StyleSheet } from 'react-native';
+import {
+  CenteredView,
+  Logger,
+  IconLoading,
+} from '@kiwicom/react-native-app-shared';
 import { connect } from '@kiwicom/react-native-app-redux';
 
 import AllHotelsSearchRow from './AllHotelsSearchRow';
-import type { AllHotelsSearchList as AllHotelsSearchListProps } from './__generated__/AllHotelsSearchList.graphql';
+import type { AllHotelsSearchList as AllHotelsSearchListProps } from './__generated__/AllHotelsSearchList_data.graphql';
 import type { CurrentSearchStats } from '../filter/CurrentSearchStatsType';
+import { HOTELS_PER_LOAD } from './AllHotelsSearch';
 
 type Props = {|
   openSingleHotel: (id: string) => void,
   data: AllHotelsSearchListProps,
   setCurrentSearchStats: (currentSearchStats: Object) => void,
+  relay: Relay,
 |};
 
-export class AllHotelsSearchList extends React.Component<Props> {
+type Hotel = {|
+  node: {
+    id: String,
+  },
+|};
+
+type State = {|
+  isLoading: boolean,
+|};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingVertical: 3,
+  },
+  loadingView: {
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingVertical: 10,
+  },
+});
+
+export class AllHotelsSearchList extends React.Component<Props, State> {
+  state = {
+    isLoading: false,
+  };
+
   componentDidMount = () => {
     Logger.LogEvent(Logger.Event.Displayed, Logger.Category.Ancillary, {
       type: 'Hotels',
       step: 'results',
     });
 
-    const currentSearchStats = idx(this.props, _ => _.data.stats);
+    const currentSearchStats = idx(
+      this.props,
+      _ => _.data.allAvailableHotels.stats,
+    );
 
     if (currentSearchStats && currentSearchStats.priceMax) {
       this.props.setCurrentSearchStats(currentSearchStats);
     }
   };
 
-  render = () => {
-    const hotels = idx(this.props, _ => _.data.edges) || [];
+  loadMore = () => {
+    if (this.props.relay.hasMore() && !this.props.relay.isLoading()) {
+      this.setState({ isLoading: true });
+      this.props.relay.loadMore(HOTELS_PER_LOAD, () => {
+        this.setState({ isLoading: false });
+      });
+    }
+  };
 
+  keyExtractor = (hotel: Hotel) => hotel.node.id;
+
+  renderListItem = ({ item }: { item: Hotel }) => (
+    <AllHotelsSearchRow
+      data={item.node}
+      openSingleHotel={this.props.openSingleHotel}
+    />
+  );
+
+  renderLoadingIndication = () => {
+    if (this.state.isLoading) {
+      return (
+        <View style={styles.loadingView}>
```

Comment:
I noticed that the loading indicator sometimes hovers over the last hotel like when you have a footer with CSS fixed position. Is it intentional?

Reply:
Though it was the only way, but with some googling I found ListFooterComponent, so I can render it at the end of the list.